tests/ui/forbid_unsafe.rs

```diff
@@ -0,0 +1,22 @@
+#![forbid(unsafe_code)]
+
```

Comment:
I wonder, does it make sense to include all the hygiene tests in here?

Something like

```rust
use pyo3::*;  // to make this `crate` match PyO3's `crate` used in the hygiene tests

#[path = "../../src/tests/hygiene/mod.rs"]
mod hygiene;
```

Reply:
Very smart! This can't test the `cfg` variants in there, but I think that's fine, so I simply allowed `unexpected_cfgs` for that.